addons/web/static/src/core/position_hook.js

```diff
@@ -259,21 +275,36 @@ export function reposition(reference, popper, options) {
  * @param {Options} options
  */
 export function usePosition(reference, options) {
-    const popper = options.popper || DEFAULTS.popper;
-    const popperRef = useRef(popper);
+    const popperRef = useRef(options.popper || DEFAULTS.popper);
     const getReference = typeof reference === "function" ? reference : () => reference;
-    const update = () => {
-        const ref = getReference();
-        if (popperRef.el && ref) {
-            reposition(ref, popperRef.el, options);
-        }
-    };
-    useEffect(update);
-    const throttledUpdate = useThrottleForAnimation(update);
-    useExternalListener(document, "scroll", throttledUpdate, { capture: true });
-    useExternalListener(window, "resize", throttledUpdate);
-    if (options.iframe) {
-        const { contentWindow: iframeWindow } = options.iframe;
-        useExternalListener(iframeWindow, "scroll", throttledUpdate, { capture: true });
-    }
+    const throttledReposition = useThrottleForAnimation(reposition);
+    useEffect(
+        (refEl, popperEl) => {
+            if (!refEl || !popperEl) {
+                return;
+            }
+
+            // Prepare
+            const refDocument = refEl.ownerDocument;
+            const iframe = getIFrame(reference);
```

Comment:
shouldn't reference be refEl here?

Makes me wondering whether getBestPosition shouldn't find the iframe itself.

Reply:
Yes it should be refEl, thanks!

At first getBestPosition was finding the iframe itself, but getBestPosition is called with the throttle on every animationFrame, so I thought it is better to compute it before attaching scroll/resize handlers below.